src/components/button/styled.tsx

```diff
@@ -0,0 +1,154 @@
+import React from "react"
+import styled, { css } from "styled-components"
+import { Button } from "@rmwc/button"
+import { CircularProgress } from "@rmwc/circular-progress"
+// import "@rmwc/circular-progress/circular-progress.css"
+import { getColor } from "../../theme/utils"
+import { ButtonType } from "./button"
+
+export const StyledButton = styled(({ label, ...otherProps }) => (
+  <Button label={label} {...otherProps} />
+))`
+  ${props => {
+    if (props.disabled) {
+      return css`
+        opacity: 0.6;
+        cursor: not-allowed;
+        background: ${getColor(["green", "greenHaze"])};
+        border-radius: 3px;
+        border-color: ${getColor(["green", "greenHaze"])};
+        width: ${props.label ? "128px" : "40px"};
+        height: 40px;
+        font-family: IBM Plex Sans;
+        font-style: normal;
+        font-weight: bold;
+        font-size: 12px;
+        color: ${getColor(["white", "pure"])};
+        &:active {
+          border-width: 0px;
+        }
+        &:focus {
+          outline: none;
+        }
+        display: ${props.icon ? "flex" : "block"};
+        flex-flow: row nowrap;
+        align-items: center;
+        .rmwc-icon {
+          margin-right: ${props.label ? "12px" : "0px"};
+          height: 24px;
+        }
+      `
+    }
+    switch (props.type) {
+      case ButtonType.noFill:
```

Comment:
There will be more button types added.

Reply:
may be it will be good idea to use to Map with fallback? to not create everytime switch and not to increase complexity of switching. Coz every new option will be checket every time in worst case. But in Map with fallback complexity is always O(1)

Reply:
Done! Thanks @builat !

Reply:
But where is `Map` based implementation?

Reply:
Its here!: https://github.com/netdata/netdata-ui/pull/3/files#diff-685bc1de267168e93c724e89256ee18d
Perhaps you see the old version in the comments? Refresh?